ram_report not working for qemu targets

commit 4bd904ccaec5ebcddfb8d9be6290facbd38beec7 introduced a bug where ram_reports are not working anymore with qemu_x86:

add CONFIG_BUILD_OUTPUT_BIN=y to the prj.conf

when building with ram_report for qemu_x86, you get:

```
Traceback (most recent call last):
  File "/work/zephyr/github/zephyr/scripts/footprint/size_report", line 420, in <module>
    base + '/', None)
  File "/work/zephyr/github/zephyr/scripts/footprint/size_report", line 194, in generate_target_memory_section
    symbols_paths = dict(load_symbols_and_paths(bin_nm, elf_file_abs, source_dir))
  File "/work/zephyr/github/zephyr/scripts/footprint/size_report", line 57, in load_symbols_and_paths
    processed_path = Path(path).relative_to(Path(path_to_strip))
  File "/usr/lib64/python3.6/pathlib.py", line 870, in relative_to
    .format(str(self), str(formatted)))
ValueError: '/usr/src/debug/libgcc/6.2.0-r0/gcc-6.2.0/build.i586-zephyr-elf.i586-zephyr-elf/libgcc/../../../../../../../work-shared/gcc-6.2.0-r0/gcc-6.2.0/libgcc/libgcc2.c' does not start with '/work/zephyr/github/zephyr'
make[4]: *** [zephyr/cmake/reports/CMakeFiles/ram_report.dir/build.make:57: zephyr/cmake/reports/CMakeFiles/ram_report] Error 1
make[3]: *** [CMakeFiles/Makefile2:1951: zephyr/cmake/reports/CMakeFiles/ram_report.dir/all] Error 2
make[2]: *** [CMakeFiles/Makefile2:1958: zephyr/cmake/reports/CMakeFiles/ram_report.dir/rule] Error 2
make[1]: *** [Makefile:495: ram_report] Error 2
make: *** [/home/nashif/bin/zb:23: ram_report] Error 2
```

--- a/scripts/footprint/size_report
+++ b/scripts/footprint/size_report
@@ -35,7 +35,13 @@ def load_symbols_and_paths(bin_nm, elf_file, path_to_strip=""):
         symbol, path = parse_symbol_path_pair(line)
 
         if path:
-            processed_path = Path(path).relative_to(Path(path_to_strip))
+            p_path          = Path(path)
+            p_path_to_strip = Path(path_to_strip)
+            try:
+                processed_path = p_path.relative_to(p_path_to_strip)
+            except ValueError as e:
+                # path is valid, but is not prefixed by path_to_strip
+                processed_path = p_path
         else:
             processed_path = Path(":")
 

